src/chocolatey/infrastructure.app/configuration/ConfigurationOptions.cs

```diff
@@ -58,28 +56,25 @@ private static IConsole Console
                                                                     Action helpMessage)
         {
             IList<string> unparsedArguments = new List<string>();
+            var optionSet = new OptionSet 
+            {{
+                "?|help|h", "Prints out the help menu.", 
+                option => configuration.HelpRequested = option != null
+            }};
 
-            // add help only once
-            if (_optionSet.Count == 0)
-            {
-                _optionSet
-                    .Add("?|help|h",
-                         "Prints out the help menu.",
-                         option => configuration.HelpRequested = option != null);
-            }

```

Comment:
I see this is number 3 in your list. We'll need a way to clear the option set explicitly. Please hook that up.